res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,1949 @@
+    celeste: 36,
+    sky: 40,
+    blue: 44,
+    purple: 48,
+    fuscia: 52,
+    magenta: 56,
+    azalea: 60,
+    salmon: 64,
+    white: 68,
+};
+
+/*
+ * USER CONFIGURABLE SETTINGS
+ * Adjust these to your liking
+ */
+
+const deckColors = [
+    LEDColors.red,
+    LEDColors.blue,
+    LEDColors.yellow,
+    LEDColors.purple,
+];
+
+const quickEffectPresetColors = [
+    null,
+    LEDColors.red,
+    LEDColors.blue,
+    LEDColors.yellow,
+    LEDColors.purple,
+
+    LEDColors.magenta,
+    LEDColors.azalea,
+    LEDColors.salmon,
+
+    LEDColors.sky,
+
+    LEDColors.celeste,
+    LEDColors.fuscia,
+
+    LEDColors.carrot,
+    LEDColors.orange,
+    LEDColors.honey,
+
+    LEDColors.lime,
+    LEDColors.aqua,
+    LEDColors.green,
+];
+
+// assign samplers to the crossfader on startup
+const samplerCrossfaderAssign = true;
+
+/*
+ * HID packet parsing library
+ */
+
+class HIDInputPacket {
+    constructor(reportId) {
+        this.reportId = reportId;
+        this.fields = [];
+    }
+
+    registerCallback(callback, byteOffset, bitOffset, bitLength, signed) {
+        if (typeof callback !== "function") {
+            throw Error("callback must be a function");
+        }
+
+        if (byteOffset === undefined || typeof byteOffset !== "number" || !Number.isInteger(byteOffset)) {
+            throw Error("byteOffset must be 0 or a positive integer");
+        }
+
+        if (bitOffset === undefined) {
+            bitOffset = 0;
+        }
+        if (typeof bitOffset !== "number" || bitOffset < 0 || !Number.isInteger(bitOffset)) {
+            throw Error("bitOffset must be 0 or a positive integer");
+        }
+
+        if (bitLength === undefined) {
+            bitLength = 1;
+        }
+        if (typeof bitLength !== "number" || bitLength < 1 || !Number.isInteger(bitOffset) || bitLength > 32) {
+            throw Error("bitLength must be an integer between 1 and 32");
+        }
+
+        if (signed === undefined) {
+            signed = false;
+        }
+
+        const field = {
+            callback: callback,
+            byteOffset: byteOffset,
+            bitOffset: bitOffset,
+            bitLength: bitLength,
+            oldData: 0
+        };
+        this.fields.push(field);
+
+        return {
+            disconnect: () => {
+                this.fields = this.fields.filter((element) => {
+                    return element !== field;
+                });
+            }
+        };
+    }
+
+    handleInput(byteArray) {
+        const view = new DataView(byteArray);
+        if (view.getUint8(0) !== this.reportId) {
+            return;
+        }
+
+        for (let field of this.fields) {
+            const numBytes = Math.ceil(field.bitLength / 8);
+            let data;
+
+            // Little endianness is specified by the HID standard.
+            // The HID standard allows signed integers as well, but I am not aware
+            // of any HID DJ controllers which use signed integers.
```

Comment:
I did implement that at first but I had no idea if it actually worked.

Reply:
Well, its already partly integrated because the registerCallback function takes a `signed` boolean. So as a consumer I'd expect there to be some support (even if experimental).

Reply:
Oops, I meant to remove that `signed` parameter.